src/services/ExperianService.ts

```diff
@@ -0,0 +1,143 @@
+import { Logger } from "@aws-lambda-powertools/logger";
+import { Constants } from "../utils/Constants";
+import axios, { AxiosRequestConfig } from "axios";
+import { AppError } from "../utils/AppError";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { ExperianTokenResponse } from "../models/IExperianResponse";
+import { sleep } from "../utils/Sleep";
+import { logResponseCode } from "../utils/LogResponseCode";
+
+export class ExperianService {
+	readonly logger: Logger;
+
+	private static instance: ExperianService;
+
+    readonly experianBaseUrl: string;
+
+    readonly backoffPeriodMs: number;
+
+    readonly maxRetries: number;
+
+    constructor(logger: Logger, experianBaseUrl: string, backoffPeriodMs: number, maxRetries: number) {
+    	this.logger = logger;
+    	this.experianBaseUrl = experianBaseUrl;
+    	this.backoffPeriodMs = backoffPeriodMs;
+    	this.maxRetries = maxRetries;
+    }
+
+    static getInstance(logger: Logger, experianBaseUrl: string, backoffPeriodMs: number, maxRetries: number): ExperianService {
+    	if (!ExperianService.instance) {
+    		ExperianService.instance = new ExperianService(logger, experianBaseUrl, backoffPeriodMs, maxRetries);
+    	}
+    	return ExperianService.instance;
+    }
+
+    // eslint-disable-next-line max-lines-per-function
+    async verify(
+    	{ accountNumber, sortCode, name, uuid }: { accountNumber: string; sortCode: string; name: string; uuid: string }, token: string,
+    ): Promise<any> {
+    	const params = {
+    		header: {
+			  tenantId: uuid,
+			  requestType: "BAVConsumer-Standard",
+    		},
+    		account: { accountNumber, sortCode },
+    		subject: { name },
+		  };
+		  
+    	const headers = {
+    		"User-Agent": Constants.EXPERIAN_USER_AGENT,
+    		"Authorization": `Bearer ${token}`,
+    		"Content-Type":"application/json",
+    		"Accept":"application/json",
+    	};
+
+    	let retryCount = 0;
+    	let exponentialBackOffPeriod = this.backoffPeriodMs;
+    	while (retryCount <= this.maxRetries) {
+    		try {
+    			const endpoint = `${this.experianBaseUrl}/${Constants.EXPERIAN_VERIFY_ENDPOINT_PATH}`;
+    			this.logger.info("Sending verify request to Experian", { uuid, endpoint, retryCount });
+    			const { data } = await axios.post(endpoint, params, { headers });
+    			const decisionElements = data?.clientResponsePayload?.decisionElements;
+
+    			this.logger.debug({
+    				message: "Recieved response from Experian verify request",
+    				eventType: decisionElements[1].auditLogs[0].eventType,
```

Comment:
I don't think we can guarantee the position in the array for auditLogs could we search for this